interview-questions.md

```diff
@@ -7,44 +8,47 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. In a model called Animal that has_many Sightings, what is the name of the foreign key? Would the foreign key be part of the Animal model or the Sightings model?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer: The foreign key would be animal_id and this would be in the Sightings model.
 
+Researched answer: Using custom association names in Rails can be a great way of increasing understanding throughout an application. For example, it can allow for using a column called owner_id that references a User record.
 
 2. Which RESTful routes must always be passed params? Why?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer: Show, Update and Delete. These routes need a specific id that is to be acted upon.
 
+Researched answer: REST APIs have several types of parameters: Header parameters: Parameters included in the request header, usually related to authorization. Path parameters: Parameters within the path of the endpoint, before the query string ( ? ). These are usually set off within curly braces.
 
 3. Name three rails generator commands. What is created by each?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer: model (used to create the model table for the database), controller (used to create the controller class for a given model), migration (used to make changes to the model <ie add columns, rename columns>)
 
+Researched answer: Others are rspec (creates new spec file for the model), resource (If you are building an API, using a front end MVC framework, or simply want to manually create your views, the resource generator is a great option for creating the code.), scaffold (Is a full set of model, database migration for that model, controller to manipulate it, views to view and manipulate the data, and a test suite for each of the above.)
 
```

Comment:
Great research again adding in the scaffold generator command even though we never learned it in class. Awesome job looking ahead